src/main/java/de/danielbechler/diff/selector/CollectionItemElementSelector.java

```diff
@@ -71,6 +94,6 @@ public boolean equals(final Object o)
 	@Override
 	public int hashCode()
 	{
-		return item != null ? item.hashCode() : 0;
+		return item != null ? identityStrategy.hashCode(item) : 0;
```

Comment:
Errrm, you mean have a constant hashcode to make sets use equals on all of them?